packages/bundler-plugin-core/src/sentry/sourcemaps.ts

```diff
@@ -1,38 +1,66 @@
 import path from "path";
-import fs from "fs";
+import fs, { Stats } from "fs";
+import glob from "glob";
+import { InternalIncludeEntry } from "../options-mapping";
 
 export type FileRecord = {
   name: string;
   content: string;
 };
 
-export function getFiles(path: string, allowedExtensions: string[]): FileRecord[] {
-  const includedFiles = getAllIncludedFileNames(path, allowedExtensions, []);
+export function getFiles(fpath: string, includeEntry: InternalIncludeEntry): FileRecord[] {
+  let fileStat: Stats;
+  const p = path.isAbsolute(fpath) ? fpath : path.resolve(process.cwd(), fpath);
+  try {
+    fileStat = fs.statSync(p);
+  } catch (e) {
+    return [];
+  }
 
-  return includedFiles.map((filename) => {
-    const content = fs.readFileSync(filename, { encoding: "utf-8" });
-    return { name: "~" + filename.replace(new RegExp(`^${path}`), ""), content };
+  let files: {
+    absolutePath: string;
+    // Contains relative paths without leading `.` (e.g. "foo/bar.js" or "asdf\\a\\b.js.map")
+    relativePath: string;
+  }[];
+
+  if (fileStat.isFile()) {
+    files = [{ absolutePath: p, relativePath: path.basename(p) }];
+  } else if (fileStat.isDirectory()) {
+    files = glob
+      .sync(path.join(p, "**"), {
+        nodir: true,
+        absolute: true,
+      })
+      .map((globPath) => ({ absolutePath: globPath, relativePath: globPath.slice(p.length + 1) }));
```

Comment:
Q: What does `globPath.slice(p.length + 1)` return?

Reply:
Ahh nevermind, I figured it out. This basically subtracts the root path `p` from the path we get from glob. Nice!